Xamarin.Forms.Core/CornerRadius.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Diagnostics;
+
+namespace Xamarin.Forms
+{
+	[DebuggerDisplay ("TopLeft={TopLeft}, TopRight={TopRight}, BottomLeft={BottomLeft}, BottomRight={BottomRight}")]
+	[TypeConverter (typeof (CornerRadiusTypeConverter))]
+	public struct CornerRadius
+	{
+		public double TopLeft { get; private set; }
+		public double TopRight { get; private set; }
+		public double BottomLeft { get; private set; }
+		public double BottomRight { get; private set; }
+
+		internal bool IsDefault
+		{
+			get { return TopLeft == 0 && TopRight == 0 && BottomLeft == 0 && BottomRight == 0; }
+		}
+
+		public CornerRadius (double uniformRadius) : this (uniformRadius, uniformRadius, uniformRadius, uniformRadius)
+		{
+		}
+
+		public CornerRadius (double topLeft, double topRight, double bottomLeft, double bottomRight)
+		{
+			TopLeft = topLeft;
+			TopRight = topRight;
+			BottomLeft = bottomLeft;
+			BottomRight = bottomRight;
+		}
+
+		public static implicit operator CornerRadius (double uniformRadius)
+		{
+			return new CornerRadius (uniformRadius);
+		}
+
+		bool Equals (CornerRadius other)
+		{
+			return TopLeft.Equals (other.TopLeft) && TopRight.Equals (other.TopRight) && BottomLeft.Equals (other.BottomLeft) && BottomRight.Equals (other.BottomRight);
+		}
+
+		public override bool Equals (object obj)
+		{
+			if (ReferenceEquals (null, obj))
+				return false;
+
+			return obj is CornerRadius && Equals ((CornerRadius)obj);
```

Comment:
`return obj is CornerRadius cornerRadius && Equals(cornerRadius);`

Reply:
Changed.